File: android/src/main/java/com/github/quadflask/react/navermap/RNNaverMapPolygonOverlay.java
package com.github.quadflask.react.navermap;

import android.content.Context;
import com.naver.maps.geometry.LatLng;
import com.naver.maps.map.overlay.PolygonOverlay;

import java.util.List;

public class RNNaverMapPolygonOverlay extends ClickableRNNaverMapFeature<PolygonOverlay> {
    public RNNaverMapPolygonOverlay(EventEmittable emitter, Context context) {
        super(emitter, context);
        feature = new PolygonOverlay();
    }

    public void setCoords(List<LatLng> coords) {
        feature.setCoords(coords);
    }

    public void setColor(int color) {
        feature.setColor(color);
    }

    public void setOutlineWidth(float widthInScreenPx) {
        feature.setOutlineWidth(Math.round(widthInScreenPx));
    }

    public void setOutlineColor(int color) {
        feature.setOutlineColor(color);
    }

    public void setHoles(List<List<LatLng>> holes) {
        feature.setHoles(holes);
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/feature/RNNaverMapPolygonOverlay.kt
package com.github.quadflask.react.navermap.feature

import android.content.Context
import com.naver.maps.geometry.LatLng
import com.naver.maps.map.overlay.PolygonOverlay
import kotlin.math.roundToInt

class RNNaverMapPolygonOverlay(emitter: EventEmittable, context: Context) : ClickableRNNaverMapFeature<PolygonOverlay>(emitter, context) {
  init {
    feature = PolygonOverlay()
  }

  fun setCoords(coords: List<LatLng?>) {
    feature.coords = coords
  }

  fun setColor(color: Int) {
    feature.color = color
  }

  fun setOutlineWidth(widthInScreenPx: Float) {
    feature.outlineWidth = widthInScreenPx.roundToInt()
  }

  fun setOutlineColor(color: Int) {
    feature.outlineColor = color
  }

  fun setHoles(holes: List<List<LatLng?>?>) {
    feature.holes = holes
  }
}
